src/test/app/AccountDelete_test.cpp

```diff
@@ -0,0 +1,699 @@
+    }
+
+    // Helper function to create a payment channel.
+    static Json::Value payChanCreate (
+        test::jtx::Account const& account,
+        test::jtx::Account const& to,
+        STAmount const& amount,
+        NetClock::duration const& settleDelay,
+        NetClock::time_point const& cancelAfter,
+        PublicKey const& pk)
+    {
+        Json::Value jv;
+        jv[jss::TransactionType] = jss::PaymentChannelCreate;
+        jv[jss::Account] = account.human();
+        jv[jss::Destination] = to.human();
+        jv[jss::Amount] = amount.getJson (JsonOptions::none);
+        jv[sfSettleDelay.jsonName] = settleDelay.count();
+        jv[sfCancelAfter.jsonName] =
+            cancelAfter.time_since_epoch().count() + 2;
+        jv[sfPublicKey.jsonName] = strHex (pk.slice());
+        return jv;
+    };
+
+public:
+
+    void testBasics()
+    {
+        using namespace test::jtx;
+
+        testcase("Basics");
+
+        Env env {*this};
+        Account const alice("alice");
+        Account const becky("becky");
+        Account const carol("carol");
+        Account const gw("gw");
+
+        env.fund(XRP(10000), alice, becky, carol, gw);
+        env.close();
+
+        // Alice can't delete her account and then give herself the XRP.
+        env (acctdelete (alice, alice), ter (temDST_IS_SRC));
+
+        // Invalid flags.
+        env (acctdelete (alice, becky),
+            txflags (tfImmediateOrCancel), ter (temINVALID_FLAG));
+
+        // Account deletion has a high fee.  Make sure the fee requirement
+        // behaves as we expect.
+        auto const acctDelFee {drops (env.current()->fees().increment)};
+        env (acctdelete (alice, becky), ter (telINSUF_FEE_P));
+
+        // Try a fee one drop less than the required amount.
+        env (acctdelete (alice, becky),
+            fee (acctDelFee - drops(1)), ter (telINSUF_FEE_P));
+
+        // alice's account is created too recently to be deleted.
+        env (acctdelete (alice, becky), fee (acctDelFee), ter (tecTOO_SOON));
+
+        // Give becky a trustline.  She is no longer deletable.
+        env (trust (becky, gw["USD"](1000)));
+        env.close();
+
+        // Give carol a deposit preauthorization, an offer, and a signer list.
+        // Even with all that she's still deletable.
+        env (deposit::auth (carol, becky));
+        std::uint32_t const carolOfferSeq  {env.seq (carol)};
+        env (offer (carol, gw["USD"](51), XRP (51)));
+        env (signers (carol, 1, {{alice, 1}, {becky, 1}}));
+
+        // Deleting should fail with TOO_SOON, which is a relatively
+        // cheap check compared to validating the contents of her directory.
+        env (acctdelete (alice, becky), fee (acctDelFee), ter (tecTOO_SOON));
+
+        // Close enough ledgers to almost be able to delete alice's account.
+        std::uint32_t const ledgerCount {
+            openLedgerSeq (env) + 256 - env.seq (alice)};
```

Comment:
This doesn't look right. The larger alice's account seq, the more ledger we should need to close. As written, that's backwards. I have a patch that has a function for what I think is the correct calculation (there are other instances of this bug in this file as well, the patch should address all of them.)